### PR TITLE
Add taxa to NCBIGene

### PR DESCRIPTION
**Pull request by gaurav**
This PR will add taxonomic identifiers to NCBIGene so we can differentiate human and other model organism genes from other cliques.

This PR adds support for extracting NCBI Taxon identifiers from the `NCBIGene/gene_info.gz`, storing it into `NCBIGene/taxa` in a simple tab-delimited format (`CURIE for NCBIGene\tCURIE for NCBITaxon`), adds a TaxonFactory for reading `taxa` files, and adds instructions for writing taxon information into compendium and synonym files.

Should help with TranslatorSRI/NameResolution#92.

In the current Babel run, we don't appear to have any genes with multiple taxa. We have 3,434,997 genes without any taxon information and 49,002,680 genes with one taxon. In compendia files, taxon information is stored per-identifier and per-clique. For example:

```json
{
  "type": "biolink:Gene",
  "identifiers": [
    {
      "i": "NCBIGene:1956",
      "l": "EGFR",
      "d": [],
      "t": [
        "NCBITaxon:9606"
      ]
    },
    {
      "i": "ENSEMBL:ENSG00000146648",
      "d": [],
      "t": []
    },
    {
      "i": "HGNC:3236",
      "l": "EGFR",
      "d": [],
      "t": []
    },
    {
      "i": "OMIM:131550",
      "d": [],
      "t": []
    },
    {
      "i": "UMLS:C1414313",
      "l": "EGFR gene",
      "d": [],
      "t": []
    }
  ],
  "taxa": [
    "NCBITaxon:9606"
  ]
}
```

In synonym files, taxon information is only stored per-clique:

```json
{
  "curie": "NCBIGene:1956",
  "names": [
    "SA7",
    "mENA",
    "ERRP",
    "EGFR",
    "ERBB",
    "HER1",
    "PIG61",
    "ERBB1",
    "NISBD2",
    "EGFR vIII",
    "EGFR gene",
    "EGFR Gene",
    "ONCOGENE ERBB",
    "SPECIES ANTIGEN 7",
    "proto-oncogene c-ErbB-1",
    "EPIDERMAL GROWTH FACTOR RECEPTOR",
    "epidermal growth factor receptor",
    "cell growth inhibiting protein 40",
    "erb-b2 receptor tyrosine kinase 1",
    "Epidermal Growth Factor Receptor Gene",
    "cell proliferation-inducing protein 61",
    "receptor tyrosine-protein kinase erbB-1",
    "epidermal growth factor receptor tyrosine kinase domain",
    "V-ERB-B AVIAN ERYTHROBLASTIC LEUKEMIA VIRAL ONCOGENE HOMOLOG",
    "avian erythroblastic leukemia viral (v-erb-b) oncogene homolog",
    "erythroblastic leukemia viral (v-erb-b) oncogene homolog (avian)",
    "epidermal growth factor receptor (avian erythroblastic leukemia viral (v-erb-b) oncogene homolog)"
  ],
  "types": [
    "Gene",
    "GeneOrGeneProduct",
    "GenomicEntity",
    "ChemicalEntityOrGeneOrGeneProduct",
    "PhysicalEssence",
    "OntologyClass",
    "BiologicalEntity",
    "ThingWithTaxon",
    "NamedThing",
    "Entity",
    "PhysicalEssenceOrOccurrent",
    "MacromolecularMachineMixin"
  ],
  "preferred_name": "EGFR",
  "shortest_name_length": 3,
  "curie_suffix": 1956,
  "taxa": [
    "NCBITaxon:9606"
  ]
}
```

Should be merged after PR #201.

### CODE REVIEW

File: src/createcompendia/drugchemical.py [hunk 1]
```diff
@@ -273,11 +273,25 @@ def build_conflation(rxn_concord,umls_concord,pubchem_rxn_concord,drug_compendiu
             x = line.strip().split('\t')
             subject = x[0]
             object = x[2]
-            #object is a PUBCHEM.  It's by definition a clique_leader.
+
             if subject in drug_rxcui_to_clique:
                 subject = drug_rxcui_to_clique[subject]
             elif subject in chemical_rxcui_to_clique:
                 subject = chemical_rxcui_to_clique[subject]
+            else:
```
Comment (cbizon) [inline src/createcompendia/drugchemical.py:281]: Is this drugchemical stuff supposed to be part of this PR?

Comment (gaurav) [reply src/createcompendia/drugchemical.py]: Nope, that was a rebasing issue! I've fixed that now.

**Review by cbizon — APPROVED** [review]
Approving, but I wonder if we should handle this more generically as a way to associate arbitrary properties with a clique.

**Comment by gaurav** [comment]
> Approving, but I wonder if we should handle this more generically as a way to associate arbitrary properties with a clique.

Good point! I'll start a discussion on that in https://github.com/TranslatorSRI/Babel/issues/237 and go ahead and merge this PR.

**Pull request merged**